test/functional/p2p_filter.py

```diff
@@ -104,6 +105,20 @@ def run_test(self):
             txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 7)
             filter_node.wait_for_tx(txid)
 
+        self.log.info('Check that request for filtered blocks is ignored if no filter is set')
+        filter_node.merkleblock_received = False
+        filter_node.tx_received = False
+        with self.nodes[0].assert_debug_log(expected_msgs=['sending inv', 'received getdata'], timeout=10):
```

Comment:
```suggestion
        with self.nodes[0].assert_debug_log(expected_msgs=['sending inv', 'received getdata']):
```

I don't really like having the debug log as a synchronisation mechanism when we have proper mininode methods like `wait_for_inv` to achieve the same.